### PR TITLE
add spmd rules for greater_than and less_than op.

### PR DESCRIPTION
**Pull request by zxcd**
### PR Category

Auto Parallel

### PR Types

New features

### Description

add spmd rules for greater_than and less_than op.
Pcard-73145

### CODE REVIEW

File: test/auto_parallel/spmd_rules/test_elementwise_rule.py [hunk 1]
```diff
@@ -559,6 +559,108 @@ def test_backward_multi_mesh_dim_broadcast(self):
...
+        self.assertEqual(infered_input_dist_attrs[0].dims_mapping, [0, -1])
+        self.assertEqual(infered_input_dist_attrs[1].dims_mapping, [0, -1])
+        self.assertEqual(infered_output_dist_attrs[0].dims_mapping, [0, -1])
+
+    def test_single_mesh_dim_broadcast_greater_than(self):
+        binary_rule = core.get_phi_spmd_rule("greater_than")
+        self.x_dist_tensor_spec.shape = [1, 64]
+        self.y_dist_tensor_spec.shape = [64]
+
+        # [-1, 0], [-1] --> [-1, 0], [-1], [-1, 0]
```
Comment (jeff41404) [inline test/auto_parallel/spmd_rules/test_elementwise_rule.py]: should be `# [-1, 0], [-1] --> [-1, 0], [0], [-1, 0]` ?

File: test/auto_parallel/spmd_rules/test_elementwise_rule.py [hunk 2]
```diff
@@ -559,6 +559,108 @@ def test_backward_multi_mesh_dim_broadcast(self):
...
+        binary_rule = core.get_phi_spmd_rule("greater_than")
+        # [0, -1], [-1, -1] --> [0, -1], [0, -1], [0, -1]
+        self.x_dist_tensor_spec.set_dims_mapping([0, -1])
+        self.y_dist_tensor_spec.set_dims_mapping([-1, -1])
+        result_dist_attrs = binary_rule.infer_forward(
+            self.x_dist_tensor_spec, self.y_dist_tensor_spec
+        )
+        infered_input_dist_attrs = result_dist_attrs[0]
+        infered_output_dist_attrs = result_dist_attrs[1]
+
```
Comment (jeff41404) [inline test/auto_parallel/spmd_rules/test_elementwise_rule.py:574]: before check value of dims_mapping of input and output, should check len(infered_input_dist_attrs) and len(infered_output_dist_attrs) first

File: test/auto_parallel/spmd_rules/test_elementwise_rule.py [hunk 3]
```diff
@@ -559,6 +559,108 @@ def test_backward_multi_mesh_dim_broadcast(self):
...
+        binary_rule = core.get_phi_spmd_rule("less_than")
+        # [0, -1], [-1, -1] --> [0, -1], [0, -1], [0, -1]
+        self.x_dist_tensor_spec.set_dims_mapping([0, -1])
+        self.y_dist_tensor_spec.set_dims_mapping([-1, -1])
+        result_dist_attrs = binary_rule.infer_forward(
+            self.x_dist_tensor_spec, self.y_dist_tensor_spec
+        )
+        infered_input_dist_attrs = result_dist_attrs[0]
+        infered_output_dist_attrs = result_dist_attrs[1]
+
```
Comment (jeff41404) [inline test/auto_parallel/spmd_rules/test_elementwise_rule.py:637]: same issue as test_single_mesh_dim_greater_than

File: test/auto_parallel/spmd_rules/test_elementwise_rule.py [hunk 4]
```diff
@@ -559,6 +559,108 @@ def test_backward_multi_mesh_dim_broadcast(self):
...
+        self.assertEqual(infered_input_dist_attrs[0].dims_mapping, [0, -1])
+        self.assertEqual(infered_input_dist_attrs[1].dims_mapping, [0, -1])
+        self.assertEqual(infered_output_dist_attrs[0].dims_mapping, [0, -1])
+
+    def test_single_mesh_dim_broadcast_less_than(self):
+        binary_rule = core.get_phi_spmd_rule("less_than")
+        self.x_dist_tensor_spec.shape = [1, 64]
+        self.y_dist_tensor_spec.shape = [64]
+
+        # [-1, 0], [-1] --> [-1, 0], [-1], [-1, 0]
```
Comment (jeff41404) [inline test/auto_parallel/spmd_rules/test_elementwise_rule.py]: should be `# [-1, 0], [-1] --> [-1, 0], [0], [-1, 0]` ?

**Review by liym27 — APPROVED** [review]
approve for PR-CI-Auto-Parallel

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by zhangting2020 — APPROVED** [review]
LGTM for PR-CI-Distribute-stable

**Pull request merged**